### PR TITLE
[Darwin] Fix kqueue/kevent bindings to actually work.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fix kqueue/kevent bindings to actually work. The P/Invoke signatures were
badly broken (missing an argument), so there's no way this could ever have
worked.

Additionally obsoletes kevent signatures that return bool (they're ignoring
vital information: how many events were actually returned from kevent), and
add overloads that do the right thing.

### CODE REVIEW

File: src/Darwin/KernelNotification.cs [hunk 1]
```diff
@@ -175,11 +179,62 @@ protected virtual void Dispose (bool disposing)
...
+				throw new ArgumentNullException (nameof (changeList));
+
+			if (eventList == null)
+				throw new ArgumentNullException (nameof (eventList));
+
+			if (changeList.Length < nChanges)
+				throw new ArgumentOutOfRangeException ("nChanges is larger than the number of elements in changeList", nameof (nChanges));
+
+			if (eventList.Length < nEvents)
+				throw new ArgumentOutOfRangeException ("nChanges is larger than the number of elements in changeList", nameof (nEvents));
```
Comment (spouliot) [inline src/Darwin/KernelNotification.cs]: copy/paste error ? should be `nEvents` in the message

File: src/Darwin/KernelNotification.cs [hunk 2]
```diff
@@ -175,11 +179,62 @@ protected virtual void Dispose (bool disposing)
...
+
+			unsafe {
+				fixed (KernelEvent *cp = &changeList [0])
```
Comment (spouliot) [inline src/Darwin/KernelNotification.cs]: incorrect, there might be 0 items in the array and the address of `[0]` won't be valid.

File: src/Darwin/KernelNotification.cs [hunk 3]
```diff
@@ -175,11 +179,62 @@ protected virtual void Dispose (bool disposing)
...
+					fixed (KernelEvent *ep = &eventList [0]) {
```
Comment (spouliot) [inline src/Darwin/KernelNotification.cs]: same

File: src/Darwin/KernelNotification.cs (hunk 2 above)
Comment (rolfbjarne) [inline src/Darwin/KernelNotification.cs]: Fixed by validating that the input array never has 0 elements.

File: src/Darwin/KernelNotification.cs (hunk 3 above)
Comment (rolfbjarne) [inline src/Darwin/KernelNotification.cs]: same

**Pull request merged**